mobile/src/main/java/rs/readahead/washington/mobile/views/activity/viewer/PDFReaderActivity.kt

```diff
@@ -181,6 +189,78 @@ class PDFReaderActivity : BaseLockActivity() {
                     false
                 }
         }
+
+        binding.pdfRendererView.recyclerView.addOnScrollListener(object :
+            RecyclerView.OnScrollListener() {
+            private val DIRECTION_NONE = -1
+            private val DIRECTION_UP = 0
+            private val DIRECTION_DOWN = 1
+            var totalDy = 0
+
+
+            var scrollDirection = DIRECTION_NONE
+            var listStatus = RecyclerView.SCROLL_STATE_IDLE
+
+            override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+                super.onScrollStateChanged(recyclerView, newState)
+                listStatus = newState
+
+                if (newState == RecyclerView.SCROLL_STATE_IDLE) {
+                    scrollDirection = DIRECTION_NONE
+                }
+
+                if ( newState == RecyclerView.SCROLL_STATE_SETTLING) {
+                    if (getDragDirection() == DIRECTION_DOWN || isOnTop()) {
```

Comment:
Maybe we should extract this logic for handling the settling state into a separate function to improve code organization and readability.